### PR TITLE
Refactor into message, peer, room

### PR DESCRIPTION
**Pull request by ryanrolds**
* Added Makefile with install, build, lint, run, and run-debug targets
* Setup and configured linter
* Updated README to use Makefile targets
* Refactored signaling handler into server, room, peer, and message structs+logic
* Setup Logrus w/ formatter that includes file, line, and function in log messages
* Consuming `ENV` env var, which controls log level (prod=info, dev=debug)

### CODE REVIEW

File: signalling/room.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+func (r *Room) MessagePeer(message Message) error {
+	peer := r.GetPeer(message.DestinationID)
+	if peer != nil {
+		log.Printf("cannot find peer %s", message.DestinationID)
+		return nil // Don't error, just skip
+	}
+
+	err := peer.SendMessage(message)
+	if err == nil {
+		log.Printf("problem setting message to peer %s", message.DestinationID)
```
Comment (Kourchenko) [inline signalling/room.go:101]: We should setup a pattern for error messages that show where the error occurred.
`log.Printf("[MessagePeer] Error in sending message to peer %s", message.DestinationID)`

File: signalling/room.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	r.rwLock.Lock()
+	defer r.rwLock.Unlock()
+
+	delete(r.peers, peer.id)
+}
+
+func (r *Room) MessagePeer(message Message) error {
+	peer := r.GetPeer(message.DestinationID)
+	if peer != nil {
+		log.Printf("cannot find peer %s", message.DestinationID)
```
Comment (Kourchenko) [inline signalling/room.go:95]: log.Printf("[MessagePeer] Error in sending message to peer %s", message.DestinationID)

File: signalling/room.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
+package signalling
+
+import (
+	"errors"
+	"log"
+	"sync"
+)
+
+var ErrPeerNotFound = errors.New("peer not found")
```
Comment (Kourchenko) [inline signalling/room.go:9]: Maybe something more descriptive of where the error occurred.
var ErrPeerNotFound = errors.New("[Standard] Peer Not Found")

File: signalling/room.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+		if err != nil {
+			log.Println(err)
+		}
+	case "icecandidate":
+		err := r.MessagePeer(message)
+		if err != nil {
+			log.Println(err)
+		}
+	default:
+		log.Printf(`unknown message type %s`, message.Type)
```
Comment (Kourchenko) [inline signalling/room.go:57]: log.Printf(`[Dispatch] Unknown message type %s`, message.Type)

File: signalling/room.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+	for _, peer := range room.peers {
+		// Don't send messages to source
+		if peer.id == message.SourceID {
+			continue
+		}
+
+		err := peer.SendMessage(message)
+		if err != nil {
+			log.Printf("problem broadcasting message to peer %s", message.DestinationID)
```
Comment (Kourchenko) [inline signalling/room.go:119]: `log.Printf("[Broadcast] Error occurred broadcasting message to peer %s", message.DestinationID))`

File: signalling/room.go (hunk 1 above)
Comment (ryanrolds) [reply signalling/room.go]: Resolving all logging comments as the func and file are included in with the logrus logs.

**Comment by ryanrolds** [comment]
A point of process, I would call the log changes feature creep and should have been in another PR.  But it wasn't a major change so it's no skin of my back. If this was for work, I would have pushed back and made ticket for adding a logger and replacing the existing log calls.

**Pull request merged**